File: 1. JPA basic/basic/src/main/java/study/jpa/basic/jpashop/domain/Album.java
package study.jpa.basic.jpashop.domain;

import javax.persistence.Entity;
import javax.persistence.Table;

/**
 * @author Joonhyuck Hyoung
 */
@Entity(name = "jpashop_album")
@Table(name = "jpashop_album")
public class Album extends Item{
    private String artist;
    private String etc;

    public String getArtist() {
        return artist;
    }

    public void setArtist(String artist) {
        this.artist = artist;
    }

    public String getEtc() {
        return etc;
    }

    public void setEtc(String etc) {
        this.etc = etc;
    }
}

File: 2. JPA deep/jpashop/src/main/kotlin/jpabook/jpashop/domain/item/Album.kt
package jpabook.jpashop.domain.item

import javax.persistence.*

/**
 * @author Joonhyuck Hyoung
 */
@Entity
@DiscriminatorValue("ALBUM")
class Album(
    val artist: String,
    val etc: String,

    name: String,
    price: Int,
    stockQuantity: Int,
) : Item(name, price, stockQuantity)
